design/feature-flags.md

```diff
@@ -0,0 +1,92 @@
+# Feature Flags
+
+Sometimes we want to introduce new breaking behavior because we believe this is
+the correct default behavior for the CDK. The problem, of course, is that
+breaking changes are only allowed in major versions and those are rare.
+
+This document describes a proposal for a pattern/mechanism called feature flags.
+It will allow us to introduce breaking behavior which is disabled by default (so
+existing projects will not be affected) but enabled automatically for new
+projects created through `cdk init`.
+
+## Approach
+
+The basic idea is that new breaking behavior will always be disabled by default
+and only enabled when a certain CDK context parameter is set. If not enabled,
```

Comment:
just for my clarification: When we say always disabled by default, it means breaking changes are always 'opt-in' and tooling like the cli could do that as a default by setting a context parameter. 

i.e An L2+ stable module couldn't opt into using new defaults but rather offer capabilities that can be requested through feature flags?

Reply:
Yes, exactly.